### PR TITLE
Fix hardcoded iterationsCount (issue #124)

### PR DESCRIPTION
**Pull request by archimed7592**

### CODE REVIEW

File: jose-jwt/jwa/Pbse2HmacShaKeyManagementWithAesKeyWrap.cs [hunk 1]
```diff
@@ -24,10 +24,18 @@ public byte[][] WrapNewKey(int cekSizeBits, object key, IDictionary<string, obje
             byte[] sharedKey = Encoding.UTF8.GetBytes(sharedPassphrase);
             byte[] algId = Encoding.UTF8.GetBytes((string)header["alg"]);
 
-            int iterationCount = 8192;
```
Comment (dvsekhvalnov) [inline jose-jwt/jwa/Pbse2HmacShaKeyManagementWithAesKeyWrap.cs:27]: let's simplify a bit. Let's keep that line with default `8192`

File: jose-jwt/jwa/Pbse2HmacShaKeyManagementWithAesKeyWrap.cs [hunk 2]
```diff
@@ -24,10 +24,18 @@ public byte[][] WrapNewKey(int cekSizeBits, object key, IDictionary<string, obje
...
+                iterationCount = Ensure.Type<int>(iterationCountObj, "Pbse2HmacShaKeyManagementWithAesKeyWrap management algorithm expectes p2c to be int.");
+            }
+            else
+            {
+                iterationCount = 8192;
+                header["p2c"] = iterationCount;
+            }
             byte[] saltInput = Arrays.Random(96); //12 bytes
 
-            header["p2c"] = iterationCount;
```
Comment (dvsekhvalnov) [inline jose-jwt/jwa/Pbse2HmacShaKeyManagementWithAesKeyWrap.cs:30]: let's keep this line too

File: jose-jwt/jwa/Pbse2HmacShaKeyManagementWithAesKeyWrap.cs [hunk 3]
```diff
@@ -24,10 +24,18 @@ public byte[][] WrapNewKey(int cekSizeBits, object key, IDictionary<string, obje
             byte[] sharedKey = Encoding.UTF8.GetBytes(sharedPassphrase);
             byte[] algId = Encoding.UTF8.GetBytes((string)header["alg"]);
 
-            int iterationCount = 8192;
+            int iterationCount;
+            if (header.TryGetValue("p2c", out var iterationCountObj))
+            {
+                iterationCount = Ensure.Type<int>(iterationCountObj, "Pbse2HmacShaKeyManagementWithAesKeyWrap management algorithm expectes p2c to be int.");
+            }
+            else
```
Comment (dvsekhvalnov) [inline jose-jwt/jwa/Pbse2HmacShaKeyManagementWithAesKeyWrap.cs:32]: and remove `else` block, keeping only `if` one.

**Comment by dvsekhvalnov** [comment]
Looks good to me in general, let's just simplify code a bit, see my comments inline.

**Comment by archimed7592** [comment]
@dvsekhvalnov , done

**Pull request merged**